modules/ocf_desktop/files/xsession/pdf-open

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash -e
+[ "$#" -ne 1 ] && { echo 'Bad arguments, trying evince...'; exec evince "$@"; }
+[ -f "$1" ] || { echo 'File does not exist, trying evince...'; exec evince "$@"; }
+
+size=$(du "$1") || { echo 'Cannot get size, trying evince...'; exec evince "$@"; }
+size=$(cut -f1 <<< "$size")
+[ "$size" -lt 5120 ] || { echo 'Too large, trying evince...'; exec evince "$@"; }
+
+echo 'Rasterizing PDF...'
+tmp=$(mktemp --suffix='.pdf')
+
+if ! convert -density 250 "$1" "$tmp"; then
+    rm "$tmp"
+    echo 'Convert failed, trying evince...'
+    exec evince "$@"
+fi
+
+echo 'Done rasterizing, now overwriting unrasterized version...'
+exiftool -tagsfromfile "$@" -title "$tmp"
```

Comment:
Check that this script only gets one argument while you're at it?

In any case you'd probably want `$1` here as `$@` is misleading.